packages/material-ui/src/Box/styleFunction.js

```diff
@@ -30,130 +26,17 @@ const filterProps = [
-    return positions(inputProps);
-  }
-  if (palette.filterProps.indexOf(prop) !== -1) {
-    return palette(inputProps);
-  }
-  if (shadows.filterProps.indexOf(prop) !== -1) {
-    return shadows(inputProps);
-  }
-  if (sizing.filterProps.indexOf(prop) !== -1) {
-    return sizing(inputProps);
-  }
-  if (spacing.filterProps.indexOf(prop) !== -1) {
-    return spacing(inputProps);
-  }
-  if (typography.filterProps.indexOf(prop) !== -1) {
-    return typography(inputProps);
-  }
-  return { [prop]: value };
-};
-
-function createEmptyBreakpointObject(breakpoints) {
-  const breakpointsInOrder = breakpoints.keys.reduce((acc, key) => {
-    const breakpointStyleKey = breakpoints.up(key);
-    acc[breakpointStyleKey] = {};
-    return acc;
-  }, {});
-  return breakpointsInOrder;
-}
-
-function removeUnusedBreakpoints(breakpointKeys, style) {
-  return breakpointKeys.reduce(
-    (acc, key) => {
-      const breakpointOutput = acc[key];
-      const isBreakpointUnused = Object.keys(breakpointOutput).length === 0;
-      if (isBreakpointUnused) {
-        delete acc[key];
-      }
-      return acc;
-    },
-    { ...style },
-  );
-}
-
-const mergeBreakpointsInOrder = (breakpoints, ...styles) => {
-  const emptyBreakpoints = createEmptyBreakpointObject(breakpoints);
-  const mergedOutput = [emptyBreakpoints, ...styles].reduce(
-    (prev, next) => deepmerge(prev, next),
-    {},
-  );
-  return removeUnusedBreakpoints(Object.keys(emptyBreakpoints), mergedOutput);
-};
-
-export const styleFunctionSx = (styles, theme) => {
-  if (!styles) return null;
-
-  if (typeof styles === 'function') {
-    return styles(theme);
-  }
-
-  if (typeof styles !== 'object') {
-    // value
-    return styles;
-  }
-
-  let css = {};
-
-  Object.keys(styles).forEach((styleKey) => {
-    if (typeof styles[styleKey] === 'object') {
-      if (filterProps.indexOf(styleKey) !== -1) {
-        css = deepmerge(css, getThemeValue(styleKey, styles[styleKey], theme));
-      } else {
-        const breakpointsValues = handleBreakpoints({ theme }, styles[styleKey], (x) => ({
-          [styleKey]: x,
-        }));
-
-        if (objectsHaveSameKeys(breakpointsValues, styles[styleKey])) {
-          const transformedValue = styleFunctionSx(styles[styleKey], theme);
-          css[styleKey] = transformedValue;
-        } else {
-          css = deepmerge(css, breakpointsValues);
-        }
-      }
-    } else if (typeof styles[styleKey] === 'function') {
-      css = deepmerge(css, { [styleKey]: styles[styleKey](theme) });
-    } else {
-      css = deepmerge(css, getThemeValue(styleKey, styles[styleKey], theme));
-    }
-  });
-  return css;
-};
-
 const styleFunction = (props) => {
   let result = {};
   Object.keys(props).forEach((prop) => {
     if (filterProps.indexOf(prop) !== -1 && prop !== 'sx') {
       result = deepmerge(result, getThemeValue(prop, props[prop], props.theme));
     }
   });
-
-  const sxValue = styleFunctionSx(props.sx, props.theme);
```

Comment:
The `experimentalStyled` already supports the sx prop, we do not need to again add it here. This was causing the same styles to be added twice cc @oliviertassinari

Reply:
Oh god, yes! So much better. The DX for debugging style was horrible without 😁.